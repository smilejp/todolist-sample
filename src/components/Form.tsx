import * as React from 'react';
import './Form.css';

interface IProps {
  value?: string;
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
  onCreate?: () => void;
  onKeyPress?: (e: React.KeyboardEvent<HTMLInputElement>) => void;
}

const Form: React.FunctionComponent<IProps> = ({ value, onChange, onCreate, onKeyPress }) => {
  return (
    <div className="form">
      <input value={value} onChange={onChange} onKeyPress={onKeyPress} />
      <div className="create-button" onClick={onCreate}>
        추가
      </div>
    </div>
  );
};

export default Form;
