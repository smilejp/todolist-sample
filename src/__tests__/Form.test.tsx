import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Form from '../components/Form';
import '../testSetup';

describe('<Form/>', () => {
  let component: ShallowWrapper<React.Component>;

  it('renders with no props', () => {
    component = shallow(<Form />);
    expect(component).toBeTruthy();
    expect(component).toMatchSnapshot();
  });
});
