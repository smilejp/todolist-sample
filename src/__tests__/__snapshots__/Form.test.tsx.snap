// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form/> renders with no props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input />,
        <div
          className="create-button"
          onClick={[Function]}
        >
          추가
        </div>,
      ],
      "className": "form",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": undefined,
          "onKeyPress": undefined,
          "value": undefined,
        },
        "ref": Object {
          "current": null,
        },
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "추가",
          "className": "create-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "추가",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input />,
          <div
            className="create-button"
            onClick={[Function]}
          >
            추가
          </div>,
        ],
        "className": "form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": undefined,
            "onKeyPress": undefined,
            "value": undefined,
          },
          "ref": Object {
            "current": null,
          },
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "추가",
            "className": "create-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "추가",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
